server.js

```diff
@@ -1,23 +1,63 @@
 const express = require("express");
 const cors = require("cors");
-
+const bodyParser = require("body-parser");
+const { request, response } = require("express");
+let changeID = 0;
 const app = express();
-
+app.use(express.json());
 app.use(cors());
-
+app.use(bodyParser.urlencoded({ extended: true }));
 const welcomeMessage = {
-  id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
 
+//This array is our "data store".
+//We will start with one message in the array.
+//Note: messages will be lost when Glitch restarts our server.
+let messages = [welcomeMessage];
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
-
-app.listen(process.env.PORT);
+app.get("/messages", function (request, response) {
+  response.send({ messages });
+});
+app.get("/messages/:id", function (request, response) {
+  const messageId = +request.params.id;
+  console.log(messageId);
+  const oneMessage = messages.find((item) => item.id === messageId);
+  response.send({ oneMessage });
+  console.log("one message:", oneMessage);
+});
+app.post("/messages", function (request, response) {
+  const { from, text } = request.body;
+  const validation =
```

Comment:
We can simplify this:
```
if (text && from) {...}
```

if text or from are undefined, they return false
if text or from are empty strings, they return false

You do not need `request.body.hasOwnProperty("text")` as you have already assigned them with `const { from, text } = request.body;`